packages/docusaurus/src/webpack/plugins/CleanWebpackPlugin.ts

```diff
@@ -147,11 +148,17 @@ export default class CleanWebpackPlugin {
    * Warning: It is recommended to initially clean your build directory outside
    * of webpack to minimize unexpected behavior.
    */
-  handleInitial(): void {
+  async handleInitial(): Promise<void> {
     if (this.initialClean) {
       return;
     }
 
+    if (await fs.pathExists(this.outputPath)) {
```

Comment:
The `outputPath` is _supposed_ to exist—this method is used to _clean_ the output. What's problematic is if it's not a directory. You also need to test if it's a file.

Reply:
Right, I will add a validation for file as well

Reply:
I have a confusion here, pathExistsSync will still return true when `build` is a file. Then why do we need an additional validation?

Reply:
There are three cases here:

1. `outputPath` doesn't exist -> Always good; `removeFiles` does nothing
2. `outputPath` is a directory -> This is why we have this function; we want to clean its content and prepare for the next build output
3. `outputPath` is a file -> This is what https://github.com/facebook/docusaurus/issues/9110 is reporting. In this case `delSync` fails because `cwd` is a file, not a directory. Logically, we can't clear anything from this file, so we fail.

We don't want to touch the behavior in cases 1 and 2, only give a better error in 3.

Reply:
I see, I misunderstood earlier. We do not need to change the behaviour when `this.outputPath` is a directory. So the check that I've put should only validate files and not directories.

Reply:
Reading the original issue, I came across another observation. the error message should occur in case-sensitive environments as well. To check if there are no existing files with same name as `build` but different case, I created a private validator that returns boolean. checks for both files and directories with duplicate name. This will solve:
 - case where root dir may have diff variations of same file such as `BUILD`, `build`, `buiLD`.
 - diff variations of same directories.

Reply:
tested the code in local env.